R/derive_vars_duration.R

```diff
@@ -143,7 +146,7 @@ derive_vars_duration <- function(dataset,
     )
 
   if (!quo_is_null(new_var_unit)) {
-    dataset <- dataset %>% mutate(!!new_var_unit := toupper(out_unit))
+    dataset <- dataset %>% mutate(!!new_var_unit := ifelse(is.na(!!new_var), NA, toupper(out_unit)))
```

Comment:
Please use `if_else` and make sure to replace `NA` with `NA_character_`.

Reply:
Updated